app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeViewModel.kt

```diff
@@ -93,24 +100,35 @@ class HomeViewModel @Inject constructor(
         if (isRefreshing) hideLoading() else _showMyCoinsLoading.value = false
     }
 
-    private fun getTrendingCoins(isRefreshing: Boolean) = execute {
-        if (isRefreshing) showLoading() else _showTrendingCoinsLoading.value = true
-        getTrendingCoinsUseCase.execute(
-            GetTrendingCoinsUseCase.Input(
-                currency = FIAT_CURRENCY,
-                order = MY_COINS_ORDER,
-                priceChangeInHour = MY_COINS_PRICE_CHANGE_IN_HOUR,
-                itemPerPage = MY_COINS_ITEM_PER_PAGE,
-                page = MY_COINS_INITIAL_PAGE
+    override fun getTrendingCoins(isRefreshing: Boolean, loadMore: Boolean) {
+        if (_showTrendingCoinsLoading.value != LoadingState.Idle) return
+        execute {
+            if (isRefreshing) showLoading() else
+                _showTrendingCoinsLoading.value = if (loadMore) LoadingState.LoadingMore else LoadingState.Loading
+            getTrendingCoinsUseCase.execute(
+                GetTrendingCoinsUseCase.Input(
+                    currency = FIAT_CURRENCY,
+                    order = MY_COINS_ORDER,
+                    priceChangeInHour = MY_COINS_PRICE_CHANGE_IN_HOUR,
+                    itemPerPage = MY_COINS_ITEM_PER_PAGE,
+                    page = trendingCoinsPage
+                )
             )
-        )
-            .catch { e ->
-                _error.emit(e)
-            }
-            .collect { coins ->
-                _trendingCoins.emit(coins.map { it.toUiModel() })
-            }
-        if (isRefreshing) hideLoading() else _showTrendingCoinsLoading.value = false
+                .catch { e ->
+                    _error.emit(e)
+                }
+                .collect { coins ->
+                    val newCoinList = coins.map { it.toUiModel() }
+                    if (isRefreshing) {
+                        _trendingCoins.emit(newCoinList)
+                    } else {
+                        _trendingCoins.emit(_trendingCoins.value + newCoinList)
+                    }
```

Comment:
@luongvo I did get this ☝️  Could you please explain to me more why we need to `_trendingCoins.value + newCoinList` 🙏

Reply:
@Wadeewee In case of loading more, we need to join the previous list to the new page list. The `emit()` func will push the new value and trigger to return the new list as state to Compose for rendering, without `_trendingCoins.value + ` we will lose the previous loaded pages.

Reply:
Ah, I see we need to handle for `loadMore` case, Thanks for your explanation @luongvo  🙏